Added Quickstart for Javascript Jobs  HTTP api

ConstantinChirila opened the pull request:
# Description

Added the missing quickstart for how to use the Dapr Jobs with Javascript through standard http fetch. 

## Issue reference

We strive to have all PRs being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: https://github.com/dapr/quickstarts/issues/1112

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [X] The quickstart code compiles correctly
* [X] You've tested new builds of the quickstart if you changed quickstart code
* [X] You've updated the quickstart's README if necessary
* [X] If you have changed the steps for a quickstart be sure that you have updated the automated validation accordingly. All of our quickstarts have annotations that allow them to be executed automatically as code. For more information see [mechanical-markdown](https://github.com/dapr/mechanical-markdown#readme). For user guide with examples see [Examples](https://github.com/dapr/mechanical-markdown/tree/main/examples#mechanical-markdown-by-example).

alicejgibbons commented on jobs/javascript/http/README.md:
```diff
@@ -0,0 +1,166 @@
+
+In this quickstart, you'll schedule, get, and delete a job using Dapr's Job API. This API is responsible for scheduling and running jobs at a specific time or interval.
+
+Visit [this](https://v1-14.docs.dapr.io/developing-applications/building-blocks/jobs/) link for more information about Dapr and the Jobs API.
```
```suggestion
Visit [this](https://docs.dapr.io/developing-applications/building-blocks/jobs/) link for more information about Dapr and the Jobs API.
```

alicejgibbons commented on jobs/javascript/http/README.md:
```diff
@@ -0,0 +1,166 @@
+
+Visit [this](https://v1-14.docs.dapr.io/developing-applications/building-blocks/jobs/) link for more information about Dapr and the Jobs API.
+
+> **Note:** This example leverages HTTP `requests` only. If you are looking for the example using the Dapr Client SDK (recommended) [click here](../sdk/).
```
```suggestion
```

alicejgibbons commented on jobs/javascript/http/README.md:
````diff
@@ -0,0 +1,166 @@
+name: Install Node dependencies
+-->
+
+```bash
````
Nit: Would break these into two command steps

alicejgibbons commented on jobs/javascript/http/README.md:
```diff
@@ -0,0 +1,166 @@
+
+<!-- END_STEP -->
+
+## Run the Jobs APIs individually
```
```suggestion
## Run the apps individually
```

alicejgibbons commented on jobs/javascript/http/job-service/package.json:
```diff
@@ -0,0 +1,18 @@
+    "type": "module",
+    "scripts": {
+        "start": "node index.js",
+        "start:dapr": "dapr run --app-port 5001 --app-id order-processing --app-protocol http --dapr-http-port 3501 --resources-path ../../../components -- npm run start"
```
Same app-id changes here - needs to be `job-service` in this case.

alicejgibbons commented on jobs/javascript/http/job-scheduler/package.json:
```diff
@@ -0,0 +1,18 @@
+{
+    "name": "order-processor",
```
`order-processor` not correct here either - this should be the name of the app-id, so `job-scheduler`

alicejgibbons commented on jobs/javascript/http/job-scheduler/package.json:
```diff
@@ -0,0 +1,18 @@
+    "type": "module",
+    "scripts": {
+        "start": "node index.js",
+        "start:dapr": "dapr run --app-port 5001 --app-id order-processing --app-protocol http --dapr-http-port 3501 --resources-path ../../../components -- npm run start"
```
change app-id also

alicejgibbons replied:
Actually this whole Dapr run command looks wrong, and will give port conflicts if run at the same time as the job-service.  Use dapr ports 3500 for one and then 3501 plus the app ports specified in the multi-app run file

alicejgibbons reviewed (changes requested):
Needs to be tested without the multi-app run files and with the single dapr commands.

ConstantinChirila commented:
> Needs to be tested without the multi-app run files and with the single dapr commands.

Thank you for the comments @alicejgibbons.
I have addressed your comments and also ran the apps with single Dapr commands.

alicejgibbons commented on jobs/javascript/http/README.md:
````diff
@@ -0,0 +1,172 @@
+```
+
+<!-- END_STEP -->
+
````
@ConstantinChirila for when you're back, I figured out that to get the `dapr stop -f .` to work you need to add `background: true` in the previous step so that it wont end the app running for you.

You can look in mine if need: https://github.com/dapr/quickstarts/pull/1144/files

ConstantinChirila replied:
@alicejgibbons  Seems to work when i validate it :)

alicejgibbons commented on jobs/javascript/http/README.md:
```diff
@@ -0,0 +1,172 @@
+expected_stderr_lines:
+output_match_mode: substring
+match_order: none
+background: false
```
```suggestion
background: true
```

alicejgibbons replied:
try this locally and lmk

marcduiker commented on jobs/javascript/http/dapr.yaml:
```diff
@@ -0,0 +1,13 @@
+version: 1
+apps:
+  - appDirPath: ./job-service/
```
```suggestion
  - appID: job-service
```

marcduiker commented on jobs/javascript/http/dapr.yaml:
```diff
@@ -0,0 +1,13 @@
+version: 1
+apps:
+  - appDirPath: ./job-service/
+    appID: job-service
```
```suggestion
    appDirPath: ./job-service/
```

marcduiker commented on jobs/javascript/http/dapr.yaml:
```diff
@@ -0,0 +1,13 @@
+    daprHTTPPort: 6280
+    schedulerHostAddress: localhost
+    command: ["npm", "run", "start"]
+  - appDirPath: ./job-scheduler/
```
```suggestion
  - appID: job-scheduler
```

marcduiker commented on jobs/javascript/http/dapr.yaml:
```diff
@@ -0,0 +1,13 @@
+    schedulerHostAddress: localhost
+    command: ["npm", "run", "start"]
+  - appDirPath: ./job-scheduler/
+    appID: job-scheduler
```
```suggestion
    appDirPath: ./job-scheduler/
```

marcduiker reviewed (commented):
Suggested small changes related to #1152

alicejgibbons commented on jobs/javascript/http/README.md:
```diff
@@ -0,0 +1,184 @@
+# Dapr Jobs
```
```suggestion
# Dapr Jobs API (JS HTTP)
```

alicejgibbons commented on jobs/javascript/http/dapr.yaml:
```diff
@@ -0,0 +1,13 @@
+    appDirPath: ./job-service/
+    appPort: 6200
+    daprHTTPPort: 6280
+    schedulerHostAddress: localhost
```
```suggestion
```

alicejgibbons replied:
We are in the process of removing this line from all the quickstarts Constantin.

alicejgibbons reviewed (approved):
LGTM, ready to merge after Elena's PR: https://github.com/dapr/quickstarts/pull/1148

msfussell reviewed (approved):
LGTM

marcduiker commented:
@holopin-bot @ConstantinChirila  Thank you! Here's a digital badge as a small token of appreciation.
